hydromt_wflow/workflows/rootzoneclim.py

```diff
@@ -0,0 +1,875 @@
+             nr_time_steps)
+            )
+       
+        # Load the potential evaporation and precipitation into memory to speed up
+        # the subsequent for loop.
+        #make sure order of coord is the same. 
+        Epdt = ds_sub.sel(forcing_type=forcing_type)["pet_mean"].dropna("time").transpose("index", "time").values
+        Pdt = ds_sub.sel(forcing_type=forcing_type)["precip_mean"].dropna("time").transpose("index", "time").values
+        # Loop through the time steps and determine the variables per time step.
+        for i in range(0, nr_time_steps):
+            if intercep_vars_sub != None:
+                #TODO: for now assumed that LAI contains monthly data, change this for future
+                month = pd.to_datetime(ds_sub.time[i].values).month
+                Imax = intercep_vars_sub["Imax"].sel(time=month)
+            # Determine the variables with a simple interception reservoir approach
+            canopy_storage[:,i] = canopy_storage[:,i] + Pdt[:,i]
+            precip_effective[:,i] = np.maximum(0, canopy_storage[:,i] - Imax)
+            canopy_storage[:,i] = canopy_storage[:,i] - precip_effective[:,i]
+            evap_interception[:,i] = np.minimum(Epdt[:,i], canopy_storage[:,i])
+            canopy_storage[:,i] = canopy_storage[:,i] - evap_interception[:,i]
+
+            # Update Si for the next time step
+            if i < nr_time_steps - 1:
+                canopy_storage[:,i+1] = canopy_storage[:,i]
+        
+        #insert in ds for the time that is available in each forcing type for precip and pet
+        time_forcing_type = ds_sub[["precip_mean", "pet_mean"]].sel(forcing_type=forcing_type).dropna("time").time
+        ds_sub["evap_interception"].loc[dict(forcing_type=forcing_type, time = time_forcing_type)] = evap_interception
+        ds_sub["precip_effective"].loc[dict(forcing_type=forcing_type, time = time_forcing_type)] = precip_effective
+        ds_sub["canopy_storage"].loc[dict(forcing_type=forcing_type, time = time_forcing_type)] = canopy_storage    
+    
+    return ds_sub
+
+
+def determine_storage_deficit(ds_sub, correct_cc_deficit):
+    """
+    Function to determine the storage deficit for every time step, subcatchment
+    location and dataset in ds_sub.
+
+    Parameters
+    ----------
+    ds_sub : xr.Dataset
+        Dataset containing the daily or higher-resolution data.
+
+    Returns
+    -------
+    ds_sub : xr.Dataset
+        Same as above, but containing the storage deficits per time step for all
+        forcing types.
+    """         
+    #make sure the order of the coordinates is always the same. 
+    # Calculate it per forcing type
+
+    ds_sub["storage_deficit"] = (
+        ("index", "forcing_type", "time"),
+        np.zeros(
+            (len(ds_sub["index"]), len(ds_sub["forcing_type"]), len(ds_sub["time"]))
+            )*np.nan
+        )
+
+    for forcing_type in ds_sub["forcing_type"].values:
```

Comment:
In this whole loop you are switching to numpy rather than xarray. Is there a specific reason? Because here you are still working with daily gridded data correct? So this will be really hard on memory

Reply:
the thing was that when working directly with xarray the code was a lot lot slower, this is why we decided with @RubenImhoff to switch to numpy

Reply:
the thing was that when working directly with xarray the code was a lot lot slower, this is why we decided with @RubenImhoff to switch to numpy

Reply:
I know... it's much faster because you load everything in memory.. and playing more with your chunksize argument / the forcing data chunksize? Because in your case you actually mind less to load full grid for each timestep and would like more much timestep or even all timesteps per grid cell? I know it's faster but I'm afraid for medium basin you may run into trouble already... did you try it for medium/large basins? Else numpy is fine but would be good if for example this still works for basins like the Rhine